### PR TITLE
add workflow for GitHub Actions

### PR DESCRIPTION
**Pull request by erossignon**

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      matrix:
+        node-version: [10.x, 14.x]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
```
Comment (robertsLando) [inline .github/workflows/workflow.yml:36]: I would replace all these lines with a single `run: |`, also did you try to keep only the npm install and npm test commands?

**Review by robertsLando — CHANGES_REQUESTED** [review]

**Comment by robertsLando** [comment]
Also remember to replace the badge in readme with the workflow badge

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+jobs:
+
+  build_on_windows:
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
```
Comment (robertsLando) [inline .github/workflows/workflow.yml:20]: Here I see a matrix but there is no one specified in this job

File: .github/workflows/workflow.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    - run: npm install
+    - run: npm test
+
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [10.x, 14.x]
```
Comment (robertsLando) [inline .github/workflows/workflow.yml:28]: Why don't include 12.x too?

**Review by robertsLando — CHANGES_REQUESTED** [review]

File: .github/workflows/workflow.yml (hunk 1 above)
Comment (jesec) [reply .github/workflows/workflow.yml:36]: Maybe consolidate them into `Install dependencies` step. 

`npm install` and `npm test` should remain separate steps.

File: .github/workflows/workflow.yml (hunk 3 above)
Comment (jesec) [reply .github/workflows/workflow.yml:28]: Yes. And, add Current (`15.x`) as well. It is important for this project to track the latest developments of Node.js.

File: .github/workflows/workflow.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+    - run: sudo apt-get update -qq
+    - run: sudo apt-get -yq --no-install-suggests --no-install-recommends --force-yes install gcc-4.8 g++-4.8
+    - run: sudo apt-get -yq --no-install-suggests --no-install-recommends --force-yes install libavahi-compat-libdnssd-dev # npm install mdns
+    - run: export CXX="g++-4.8" CC="gcc-4.8"
```
Comment (jesec) [inline .github/workflows/workflow.yml:40]: GCC 4.8? Not acceptable in 2021. 

Why not simply use `build-essential`?

I would also like to see a workflow for the latest GCC and Clang.**

** I can do this myself later, though.

**Comment by leerob** [comment]
Thanks for this! Looking great - agree with the feedback above 🚀

**Comment by robertsLando** [comment]
Supersided by #1074

**Comment by erossignon** [comment]
Thank you guy, let's close this one.

**Pull request closed without merging**